src/core/InstrumentSoundShaping.cpp

```diff
@@ -275,19 +258,23 @@ void InstrumentSoundShaping::processAudioBuffer( SampleFrame* buffer,
 
 f_cnt_t InstrumentSoundShaping::envFrames( const bool _only_vol ) const
 {
-	f_cnt_t ret_val = m_envLfoParameters[static_cast<std::size_t>(Target::Volume)]->PAHD_Frames();
+	f_cnt_t ret_val = getVolumeParameters().PAHD_Frames();
 
-	if( _only_vol == false )
+	if (!_only_vol)
 	{
-		for (auto i = static_cast<std::size_t>(Target::Volume) + 1; i < NumTargets; ++i)
+		auto& cutoffParameters = getCutoffParameters();
+		if (cutoffParameters.isUsed())
```

Comment:
I'm not sure what the advantage is to using getters internally within the class to access its own members. It just seems to increase verbosity compared to using member variables directly. (The const overloads of the getters can be removed too if this is changed.)

Reply:
I think this is a leftover from the incremental refactoring when the retrieval of the parameters was abstracted in these methods while still using the old array implementation.

However, I don't think it's really bad because the advantage is that it hides the implementation details of the retrieval/storage, i.e. the class does not have to care where the parameters come from or how they are managed. If I'd pepper the code with direct member accesses now it would feel like a degradation to me so I'd like to keep this as is.